conference.js

```diff
@@ -1344,6 +1344,7 @@ export default {
                 }
                 this.setVideoMuteStatus(this.isLocalVideoMuted());
                 APP.UI.updateDesktopSharingButtons();
+                APP.API.notifyScreenSharingStatusChanged(this.isSharingScreen);
```

Comment:
I'm not sure how I feel about the notification happening with every successful useVideoStream call, maybe only because the method has the words "notify" and "Changed" but gets called when maybe it hasn't really changed, such as when switching camera devices. useVideoStream does look like a pretty decent place to shove the call though. @hristoterezov, what are your thoughts?

Reply:
Agree. And probably `APP.UI.updateDesktopSharingButtons()` should also not be called each time. I think real change can be detected as follows: 
```javascript
useVideoStream(newStream) {
  const oldStream = this.localVideo;

  return APP.store.dispatch(
      // ...
      if (oldStream.videoType != newStream.videoType) {
        APP.UI.updateDesktopSharingButtons();
        APP.API.notifyScreenSharingStatusChanged(this.isSharingScreen);
      }
```

This will break though if current [VIDEO_TYPE](https://github.com/jitsi/jitsi-meet/blob/85a168d51ba82c2bd7fe6a4c2c490804cf9eccd7/react/features/base/media/constants.js#L41) enum will be expanded. If it is possible then the check should be `oldStream.videoType != newStream.videoType && (oldStream.videoType == 'desktop' || newStream.videoType == 'desktop')`